avocado/utils/config_change_validation.py

```diff
@@ -0,0 +1,205 @@
+"""
+
+SUMMARY
+------------------------------------------------------
+Config change validation: Utility for testing config file changes.
+
+Copyright: 2012 Intra2net AG
+
+
+INTERFACE
+------------------------------------------------------
+
+"""
+
+import os
+import shutil
+
+# avocado utilities
+from avocado.utils import process
+
+
+def get_temp_file_path(file_path):
+    """
+    Generates a temporary filename.
+
+    :param str file_path: file path prefix
+    :returns: appended file path
+    :rtype: str
+    """
+    return file_path + '.tmp'
+
+
+def make_temp_file_copies(file_paths):
+    """
+    Creates temporary copies of the provided files.
+
+    :param file_paths: copied file paths
+    :type file_paths: [str]
+    """
+    for file_path in file_paths:
+        temp_file_path = get_temp_file_path(file_path)
+        shutil.copyfile(file_path, temp_file_path)
+
+
+def del_temp_file_copies(file_paths):
+    """
+    Deletes all the provided files.
+
+    :param file_paths: deleted file paths (their temporary versions)
+    :type file_paths: [str]
+    """
+    for file_path in file_paths:
```

Comment:
If you go with the temporary directory approach mentioned earlier, which could be shared for all the copied temporary files, this operation could be a simple `shutil.rmtree()`.